### PR TITLE
Start Geth with debug api endpoint 

### PR DESCRIPTION
**Pull request by zsluedem**
## Solution
Introduce additional config to enable debug api endpoint.

## PR Checklist

-   [ ] Added Tests
-   [ ] Added Documentation
-   [ ] Updated the changelog
-   [ ] Breaking changes

### CODE REVIEW

File: ethers-core/src/utils/geth.rs [hunk 1]
```diff
@@ -364,7 +372,8 @@ impl Geth {
         // Open the HTTP API
         cmd.arg("--http");
         cmd.arg("--http.port").arg(port.to_string());
-        cmd.arg("--http.api").arg(API);
+        let apis = if self.enable_debug { format!("{},debug", API) } else { API.to_owned() };
+        cmd.arg("--http.api").arg(apis);
```
Comment (mattsse) [inline ethers-core/src/utils/geth.rs:376]: I would like there to be a `Vec<String>` for all additional modules. so we can also enable admin for example.

Should this also be enabled for `ws.api`?

**Review by mattsse — CHANGES_REQUESTED** [review]

File: ethers-core/src/utils/geth.rs (hunk 1 above)
Comment (zsluedem) [reply ethers-core/src/utils/geth.rs:376]: Yes. You are right.

Comment (zsluedem) [reply ethers-core/src/utils/geth.rs:376]: > I would like there to be a `Vec<String>` for all additional modules. so we can also enable admin for example.

Do you want me to add another method like `enable_admin`?

Comment (zsluedem) [reply ethers-core/src/utils/geth.rs]: @mattsse I made some changes. Take a look when you are available. Thanks.

**Review by mattsse — APPROVED** [review]
lgtm, pending q regarding trailing comma

File: ethers-core/src/utils/geth.rs [hunk 2]
```diff
@@ -364,7 +380,15 @@ impl Geth {
...
         cmd.arg("--http.port").arg(port.to_string());
-        cmd.arg("--http.api").arg(API);
+        let mut additional_apis = vec![];
+        if self.enable_debug {
+            additional_apis.push("debug".to_string())
+        };
+        if self.enable_admin {
+            additional_apis.push("admin".to_string())
+        };
+        let enable_apis = format!("{},{}", API, additional_apis.join(","));
```
[the remark this answers is not in the document]
Comment (zsluedem) [reply ethers-core/src/utils/geth.rs:390]: Good catch. I didn't notice that. I just checked with geth and it worked fine with trailing comma.

File: ethers-core/src/utils/geth.rs [hunk 3]
```diff
@@ -21,7 +21,7 @@ const GETH_STARTUP_TIMEOUT_MILLIS: u64 = 10_000;
 const GETH_DIAL_LOOP_TIMEOUT: Duration = Duration::new(20, 0);
 
 /// The exposed APIs
-const API: &str = "eth,net,web3,txpool,admin,personal,miner";
+const API: &str = "eth,net,web3,txpool,personal,miner";
```
Comment (zsluedem) [inline ethers-core/src/utils/geth.rs:24]: @mattsse I forgot to remove the default admin here. This would change the default behaviour.

Comment (mattsse) [reply ethers-core/src/utils/geth.rs:24]: oh my bad didn't realise this was in by default,

sorry about the confusion.

we want to keep admin by default. so the enable_admin should be removed, but we can keep the Vec, so custom namespaces can be enabled.

**Review by gakonst — COMMENTED** [review]
no disagreement beyond not changing the defaults

**Pull request merged**